### PR TITLE
Sb 15141 Draft: sending email in notification job

### PR DESCRIPTION
**Pull request by Hari-stackroute**

### CODE REVIEW

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java [hunk 1]
```diff
@@ -31,66 +37,80 @@
...
+				Logger.info("NotificationService:processMessage action name is incorrect: " + actionValue + "for message id:"+msgId);
 			}
 		} else {
-			Logger.info("NotificationService:processMessage event data map is either null or empty");
+			Logger.info("NotificationService:processMessage event data map is either null or empty for message id:"+msgId);
 		}
 	}
 
+	private boolean sendEmailNotification(Map<String, Object> notificationMap) {
+		List<String> deviceIds = (List<String>) notificationMap.get(Constant.IDS);
```
Comment (manzarul) [inline samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: deviceIds to emailIds

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java [hunk 2]
```diff
@@ -31,66 +37,80 @@
...
+		Map<String, Object> templateMap = (Map<String, Object>) notificationMap.get(Constant.TEMPLATE);
+		Map<String, Object> config =  (Map<String, Object>) notificationMap.get(Constant.CONFIG);
+		String subject = (String) config.get(Constant.SUBJECT);
+		String emailText = (String) templateMap.get(Constant.DATA);
+		EmailRequest emailRequest = new EmailRequest(subject, deviceIds,null,null,"", emailText,null);
+		return emailService.sendEmail(emailRequest);
+	}
+
 	private boolean sendSmsNotification(Map<String, Object> notificationMap, String msgId) {
 		List<String> deviceIds = (List<String>) notificationMap.get(Constant.IDS);
```
Comment (manzarul) [inline samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: deviceIds to phoneNumbers

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java [hunk 3]
```diff
@@ -31,66 +37,80 @@
...
-		mandatoryParams = new String[]{
-						Constant.ACTION_NAME, Constant.IDS, Constant.EDATA
-		};
-		accountKey = appConfig.get(FCM_ACCOUNT_KEY);
-		msgAuthKey = appConfig.get(MSG_AUTH_KEY);
-		SMSConfig smsConfig = new SMSConfig(msgAuthKey, "");
+		accountKey = appConfig.get(NotificationEnum.fcm_account_key.name());
+		smsAuthKey = appConfig.get(NotificationEnum.sms_auth_key.name());
+		smsDefaultSender = appConfig.get(NotificationEnum.sms_default_sender.name());
+		SMSConfig smsConfig = new SMSConfig(smsAuthKey, smsDefaultSender);
```
Comment (manzarul) [inline samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: new SMSConfig(smsAuthKey, smsDefaultSender); to new SMSConfig(appConfig.get(NotificationEnum.sms_auth_key.name()), appConfig.get(NotificationEnum.sms_default_sender.name()));

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java [hunk 4]
```diff
@@ -31,66 +37,80 @@
 	private JobLogger Logger = new JobLogger(NotificationService.class);
 	private Config appConfig = null;
 	private String accountKey = null;
-	private String msgAuthKey = null;
+	private String smsAuthKey = null;
```
Comment (manzarul) [inline samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: remove smsAuthKey and smsDefaultSender

Comment (Hari-stackroute) [reply samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: Done

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java (hunk 3 above)
Comment (Hari-stackroute) [reply samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: Done

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java (hunk 2 above)
Comment (Hari-stackroute) [reply samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: changed to mobileNumbers

File: samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java (hunk 1 above)
Comment (Hari-stackroute) [reply samza/notification-job/src/main/java/org/sunbird/jobs/samza/service/NotificationService.java]: Done

**Review by manzarul — APPROVED** [review]

**Pull request merged**